osu.Game/Skinning/SkinnableSound.cs

```diff
@@ -45,33 +52,23 @@ public bool Looping
 
                 looping = value;
 
-                channels?.ForEach(c => c.Looping = looping);
+                samplesContainer.ForEach(c => c.Looping = looping);
             }
         }
 
-        public void Play() => channels?.ForEach(c => c.Play());
-
-        public void Stop() => channels?.ForEach(c => c.Stop());
-
-        public void AddAdjustment(AdjustableProperty type, BindableDouble adjustBindable)
+        public void Play() => samplesContainer.ForEach(c =>
         {
-            if (adjustments == null) adjustments = new List<(AdjustableProperty, BindableDouble)>();
+            if (c.AggregateVolume.Value > 0)
+                c.Play();
```

Comment:
So this is a little weird to me - it's a tiny edge-case for sure, but if you did (for whatever reason)

```csharp
skinnableSound.Volume.Value = 0;
skinnableSound.Play();
skinnableSound.Volume.Value = 0.5; // assume this is called before the sample should be done
```

the sound wouldn't play at all. It's just something I'd personally consider unexpected behaviour if that is the case (I'd probably expect for it to be running "in the background", as it does without the guard).

Reply:
Same case for samples that set looping and adjust the volume after playing:
```csharp
skinnableSound.Looping = true;
skinnableSound.Volume.Value = 0;
skinnableSound.Play();
skinnableSound.TransformBindableTo(Volume, 0.75f, 250); // adjust volume.
```

which sounds like something I would expect from people to do in their rulesets

Reply:
rulesets should not be modifying the internal volume of hitsounds. they are set by the mapper.